base_layer/wallet/src/base_node_service/config.rs

```diff
@@ -41,7 +41,7 @@ impl Default for BaseNodeServiceConfig {
     fn default() -> Self {
         Self {
             base_node_monitor_max_refresh_interval: Duration::from_secs(30),
-            base_node_rpc_pool_size: 10,
+            base_node_rpc_pool_size: 3,
             event_channel_size: 250,
```

Comment:
is this perhaps not too low?
I think a wallet can have the following running at a single time:

- TMS validation
- OMS validation
- base node status check
- base node tip
- UTXO scanning

I am not sure if I miss any here.

Reply:
RPC is designed to share clients, basically you'll just have to wait for the previous request to complete before yours is started. Since this I've changed the behaviour to drop RPCs once they are not being used so perhaps a good number is the number of services using it - by your count  6. 
Isnt "base node tip" using messaging protocol through the liveness service and not RPC? Will audit this  can come up with a minimal number.

Reply:
Seems it's used in 8 places, difficult to know how many are required at the same time. In my rather basic testing it never got above 3 and if it is ever a problem in any of the services to wait some bounded time for responses. To be safe I'll set it to 8.